src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -271,16 +271,24 @@ trait GraphDAO {
   }
 
   def saveObject(tpe: Type, obj: DomainObject, vertex: Vertex, wsc: WorkspaceContext, graph: Graph): Unit = {
+    //Bit of a hack: always make sure that the idField is the first thing to be serialized.
+    //This makes sure that half-serialized objects are always findable even if they didn't save successfully.
+    saveProperty(tpe, obj.idField, getDomainObjectIdField(tpe, obj), vertex, wsc, graph)
```

Comment:
This makes sure that `saveObject` always leaves the vertex in a findable state, which is what allows me to get away with doing [this](https://github.com/broadinstitute/rawls/pull/114/files?diff=split#diff-f49c3687908d541f0fd60b9be8c20db4R107).